base-deploy

```diff
@@ -1,12 +1,7 @@
 #!/usr/bin/env bash
-set -ex
 
-export ETH_GAS=${ETH_GAS:-"7000000"}
-unset SOLC_FLAGS
-
-# Get config variables
-test -z "$CONFIG_STEP" && exit 1
-CONFIG_FILE="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )/${CONFIG_STEP}.json"
+# shellcheck source=lib/common.sh
+. "${LIB_DIR:-$(cd "${0%/*}/lib"&&pwd)}/common.sh"
```

Comment:
I guess you could use `echo` here instead of `cd && pwd`?

If you tell me what the value of `$0` is here I could help find a way to do this.

Reply:
This pattern is to avoid using `realpath` which is platform dependent. Essentially it takes the first argument passed to `bash` i.e. the path to the current script and gets the directory of that path. The reason for the `cd && pwd` is to resolve the absolute path of that directory without using `realpath`.

Reply:
Can't we solve the platform-dependence by using the `coreutils` version of `realpath`?

Reply:
This is to not break it for people not running Nix.

Reply:
Hah. As I said, I think we should eventually settle on one approach :)

But this specific case is just a visual annoyance rather than an actual issue, so it's all good.